allauth/socialaccount/providers/ebay/views.py

```diff
@@ -0,0 +1,65 @@
+# -*- coding: utf-8 -*-
+import requests
+
+from django.conf import settings
+
+# from allauth.socialaccount import app_settings
+from allauth.socialaccount.providers.ebay.provider import EBayProvider
+from allauth.socialaccount.providers.oauth2.views import (
+    OAuth2Adapter,
+    OAuth2CallbackView,
+    OAuth2LoginView,
+)
+
+ENVIRONMENTS = {
+    "production": {
+        "auth_url": "https://auth.ebay.com/oauth2/authorize",
+        "token_url": "https://api.ebay.com/identity/v1/oauth2/token",
+    },
+    "sandbox": {
+        "auth_url": "https://auth.sandbox.ebay.com/oauth2/authorize",
+        "token_url": "https://api.sandbox.ebay.com/identity/v1/oauth2/token",
+    },
+}
+
+ENV = (
+    getattr(settings, "SOCIALACCOUNT_PROVIDERS", {})
+    .get("ebay", {})
+    .get("ENVIRONMENT", "production")
```

Comment:
Whether the production or sandbox environment is used is actually a setting that we can store/tweak per app. I think it would be best to retrieve it from the `SocialApp.settings`:

``` python
  SOCIALACCOUNT_PROVIDERS = {
      "ebay": {
          "APPS": [
              {
                  "client_id": "<insert-id>",
                  "secret": "<insert-secret>",
                  "settings": {
                      "environment": "production",
                  }
              }
          ]
      }
  }
```

(note: the lower case "environment" is intentional here)

Reply:
Sounds great! Where should this change go? I can't seem to find the file. Should it be within the test_settings.py? Also is there a Contributing document I should have a look at or some way to test it? I'm struggling to know if I am on the right track.

Reply:
In your adapter class, you can do:

```
app = self.get_provider().app
environment = app.settings.get("environment", "production")
```

So, this is a change to your adapter class, not related to `test_settings.py`.